apiserver/common/leadership_test.go

```diff
@@ -103,17 +104,33 @@ func (s *LeadershipSuite) TestUnpinError(c *gc.C) {
 	c.Check(results[0].Error, gc.ErrorMatches, "boom")
 }
 
+func (s *LeadershipSuite) TestPermissionDenied(c *gc.C) {
+	s.tag = names.NewUserTag("some-random-cat")
+	defer s.setup(c).Finish()
+
+	arg := params.PinLeadershipBulkParams{
+		Params: []params.PinLeadershipParams{{
+			ApplicationTag: "application-redis",
+		}},
+	}
+	_, err := s.api.UnpinLeadership(arg)
+	c.Assert(err, gc.ErrorMatches, "permission denied")
```

Comment:
Do you also test Pin as well as Unpin?

Reply:
If we changed what machines have access to, we'd also obviously want to extend the test suite for those cases.